src/pudl/analysis/pudl_models.py

```diff
@@ -28,6 +45,176 @@ def _year_quarter_to_date(year_quarter: pd.Series) -> pd.Series:
     return pd.PeriodIndex(year_quarter, freq="Q").to_timestamp()
 
 
+def _get_cik_from_filename(filename_sec10k: pd.Series) -> pd.Series:
+    """Get the CIK of the filer from the filename strings."""
+    return filename_sec10k.str.split("/").str[2].str.zfill(10)
+
+
+def _get_sec_state_code_dict() -> dict[str, str]:
```

Comment:
Pulled in from the mozilla repo. There are a bunch of strange state codes in the SEC data that can be mapped to full location strings.